faunadb-scala/src/main/scala/faunadb/FaunaClient.scala

```diff
@@ -248,50 +240,89 @@ class FaunaClient private (connection: Connection) {
   def syncLastTxnTime(timestamp: Long): Unit =
     connection.syncLastTxnTime(timestamp)
 
-  private def handleNetworkExceptions[A]: PartialFunction[Throwable, Future[A]] = {
-    case ex: ConnectException => Future.failed(new UnavailableException(ex.getMessage, ex))
-    case ex: TimeoutException => Future.failed(new TimeoutException(ex.getMessage))
+  private def parseResponseBody(response: FullHttpResponse)(implicit ec: ExecutionContext): Future[JsonNode] = {
+    def parse: Future[Option[JsonNode]] = Future(Option(json.readTree(new ByteBufInputStream(response.content()))))
```

Comment:
is this inner method necessary? looks like you can just create an option and match on it
```scala
Option(json.readTree(new ByteBufInputStream(response.content()))) match {
  case Some(json) => Future.successful(json)
  case None       => Future.failed((new IOException("Invalid JSON."))
}
```

Reply:
I wanted the _parsing_ to happen asynchronously. So, in theory the equivalent should be the following:

```
Future(Option(json.readTree(new ByteBufInputStream(response.content())))).flatMap {
  case Some(json) => Future.successful(json)
  case None => Future.failed(new IOException("Invalid JSON."))
}
```

But it kind of feels [too much in one expression](https://gist.github.com/lregnier/57179e016fad0028964bcfd9951e33cc#2-dont-pack-too-much-in-one-expression) to me, so I used the nested method! :)